ui-particles-angular/projects/ui-particles-angular/src/lib/gio-license/gio-license.service.ts

```diff
@@ -105,11 +105,12 @@ export class GioLicenseService {
     if (!licenseOptions.feature) {
       throw new Error(`feature is undefined`);
     }
-    let url = `${this.licenseConfiguration.trialResourceURL}?utm_source=${this.licenseConfiguration.utmSource}&utm_medium=${licenseOptions.feature}&utm_campaign=${this.licenseConfiguration.utmCampaign}`;
-    if (licenseOptions.context) {
-      url += `&utm_content=${licenseOptions.context}`;
-    }
-    return url;
+    const urlParams =
+      this.licenseConfiguration.utmSource || this.licenseConfiguration.utmCampaign
+        ? `?utm_source=${this.licenseConfiguration.utmSource}&utm_medium=${licenseOptions.feature}&utm_campaign=${this.licenseConfiguration.utmCampaign}`
```

Comment:
Maybe to a check for each parameter? This would avoid `?utm_XX=undefined` if the customization doesn't provide a value.

Reply:
This will be used only by APIM and adding null check would also imply extra complexity when building the URL so after discussion, we'll leave it this way.